restrict_download_region/restrict_region.py

```diff
@@ -0,0 +1,141 @@
+
+    # generate new policy statement based on data from AWS
+    http = urllib3.PoolManager()
+    resp = http.request('GET', 'https://ip-ranges.amazonaws.com/ip-ranges.json')
+    all_ip_prefixes = json.loads(resp.data.decode('utf-8'))
+    return ip_prefixes_for_region(all_ip_prefixes['prefixes'], 'ip_prefix', AWS_REGION) + \
+        ip_prefixes_for_region(all_ip_prefixes['ipv6_prefixes'], 'ipv6_prefix', AWS_REGION)
+
+
+def get_bucket_policy(s3_client, bucket_name: str) -> dict:
+    try:
+        return json.loads(s3_client.get_bucket_policy(Bucket=bucket_name)['Policy'])
+    except botocore.exceptions.ClientError as e:
+        if e.response['Error']['Code'] == 'NoSuchBucketPolicy':
+            return {
+                "Version": "2012-10-17",
+                "Statement": []
+            }
+        else:
+            raise
+
+
+def process_ip_restrict_policy(bucket_name: str,
+                               region_ip_prefixes: List[str],
+                               custom_resource_request_type: str,
+                               bucket_policy: dict):
+    """
+    Modifies the passed in bucket_policy and decides whether
+    to add or remove the IP restriction policy
+    """
+    # filter out the previously set IP filtering policy
+    bucket_policy['Statement'] = [statement for statement in bucket_policy['Statement']
+                                  if (POLICY_STATEMENT_ID != statement.get("Sid"))]
+
+    # skip adding new policy if deleting the custom resource
+    if custom_resource_request_type == 'Delete':
+        return
+
+    # add new IP address policy statement
+    new_ip_policy_statement = generate_ip_address_policy(bucket_name, region_ip_prefixes)
+    bucket_policy['Statement'].append(new_ip_policy_statement)
+
+
+def update_bucket_policy(s3_client, bucket_name: str, bucket_policy: dict):
+    # update existing policy or delete policy completely if there are no longer any policies left
+    if bucket_policy['Statement']:
+        s3_client.put_bucket_policy(Bucket=bucket_name, Policy=json.dumps(bucket_policy))
+    else:
+        s3_client.delete_bucket_policy(Bucket=bucket_name)
+
+
+def ip_prefixes_for_region(ip_ranges: List[Dict], prefix_key: str, region: str) -> List[str]:
+    return [item[prefix_key] for item in ip_ranges if (
+        item["service"] == "AMAZON" and item["region"] == region)]
+
+
+@contextmanager
+def handle_custom_resource_status_message(event: dict, context: dict):
+    custom_resource_request_type = event.get('RequestType')
+    try:
+        yield custom_resource_request_type
+
+        # when custom_resource_request_type is None, this lambda is being
+        # triggered by Amazon's SNS topic because IP prefixes have updated
+        if custom_resource_request_type:
+            cfnresponse.send(event, context, cfnresponse.SUCCESS, {'Data': ''})
+        else:
+            print("was not a custom resource. No messages sent")
```

Comment:
how about use a logger instead?  you can look at our existing lambda for examples, i think all of them use a logger.